node/signer_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package node
+
+import (
+	"log"
+	"testing"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/config"
+)
+
+func TestDefaultConfigInitializationUsesExistingDefaultKey(t *testing.T) {
+	t.Setenv("HOME", t.TempDir())
+
+	require := require.New(t)
+	v := setupViperFlags()
+
+	config1, err := config.GetNodeConfig(v)
+	require.NoError(err)
+	signer1, err := NewStakingSigner(config1.StakingSignerConfig)
+	require.NoError(err)
+	signer2, err := NewStakingSigner(config1.StakingSignerConfig)
```

Comment:
This feels like we're testing implementation details, since this code is not a part of the public node api. I think we should just create the node through `node.New`, pass in the configuration, and do a check against the `Signer` exposed on it.

Reply:
I've simplified this test to hopefully make things more clear. What we're testing is that when a signer is created with a new key, that that key is saved and used the next time a signer is instantiated.

Reply:
My concern isn't that the test is clear - it's that we're testing implementation details. ref https://github.com/ava-labs/avalanchego/pull/3965#discussion_r2192698611